packages/matter.js/InterfaceNotes.ts

```diff
@@ -0,0 +1,121 @@
+// We agreed that server interface is the priority...  Here's a more detailed
+// proposal on the API.
+//
+// Note 1: This is a conceptual interface that models what the current PR is
+// generating (or will generate) for a single cluster.  I pulled it together
+// into a single place for discussion purposes.
+//
+// Question 1. We just call this "Device" but it's really a "ServerDevice".
+// In a controller we could potentially also have a "ClientDevice" that looks
+// very similar but proxies over the fabric to a remote endpoint.  So should
+// we consider making space in the API for a future Client/Server bifurcation
+// or am I thinking about this wrong?
+//
+// Note 2. The concept of a "ClientDevice" wouldn't be just for controllers, it
+// would also be for client clusters of a "ServerDevice".  I think.
+
+// ATTRIBUTES VALUES
+//
+// Attribute values are stored in an immutable "State" object.  This is not
+// necessarily the "source of truth".  It's just what the implementation
+// has conveyed to matter.js as the current attribute values.
+//
+// Note 3: (no longer relevant)
+//
+// Question 2: (resolved myself)
+type State = {
+    readonly currentLevel: number | undefined;
+    readonly minLevel?: number;
+    readonly maxLevel?: number;
+    readonly onOffTransitionTime?: number;
+    readonly onLevel: number | undefined;
+    readonly onTransitionTime?: number | undefined;
+    readonly offTransitionTime?: number | undefined;
+    readonly defaultMoveRate?: number | undefined;
+    readonly options: number;
+    readonly remainingTime?: number;
+    readonly startUpCurrentLevel?: number | undefined;
+}
+
+export interface PumpDevice extends Device {
+    // ATTRIBUTE MUTATION
+    //
+    // This is how the device implementation tells matter.js that attribute
+    // values are different.  Internal implementation would then read all
+    // attribute values and push any changes into the appropriate
+    // ClusterServer.
+    //
+    // Question 3: One reason I like this approach is I worry that notifying
+    // matter.js for every change would get too chatty.  Right now changes
+    // are coalesced in SubscriptionHandler before notifying subscribers so
+    // I don't think this is an issue from a network perspective.  But there
+    // are a lot of internal function calls for every attribute on every change
+    // I don't think there's any immediate need for change but is it worth
+    // considering for the shape of the public API for when there are hundreds
+    // of endpoints and thousands of attributes on a low-power device?
+    //
+    // Note 4: (no longer relevant)
+    set(changes: Partial<State>): void;
+
+    // This is a convenience form of set for updating a single attribute.
+    //
+    // Note 8: We could also generate a setter for every individual attribute.
+    // However, I worry that would encourage inefficient implementation.
```

Comment:
If there are a lot of attribute changes then we'll trigger a *lot* of callbacks.  I'm always nervous about that as it can add up and slam you with latency.  A lot of my home IOT stuff takes forever to start up which is annoying.  I've got 60+ zwave switches plus various sensors and shades.

Don't want to optimize prematurely, I just didn't want to code into a corner with public API.  But I think I can implement to current approach and keep more efficient path open for the future if it becomes necessary.

I just finished wiring for the generated attributes.  I made state attributes writable and have each read/write percolate down into the cluster server.  I'll remove the write portion for fixed attributes once your PR with appropriate metadata gets merged.